saleor/graphql/core/fields.py

```diff
@@ -186,9 +187,13 @@ def connection_resolver(
         filter_input = args.get(filters_name)
 
         if filter_input and filterset_class:
-            iterable = filterset_class(
+            instance = filterset_class(
                 data=dict(filter_input), queryset=iterable, request=info.context
-            ).qs
+            )
+            # Make sure filter input has valid values
+            if not instance.is_valid():
+                raise ValidationError(instance.errors, code="invalid_list")
```

Comment:
Please check whether that error is handled or not by GraphQL.

In order to achieve this, trigger the error and look at the server logs, if you see something saying the error was not handled, "query failed unexpectedly" it means you have to raise a `GraphQLError` instead:
```
ERROR saleor.graphql.errors.unhandled A query failed unexpectedly [PID:18114:Thread-3]
Traceback (most recent call last):
```

Reply:
I was going to write, that we had discussion about what type of errors should we return in queries and it seems that in this case `GraphQLError` is the best one. Validation errors are rather for mutations where errors can be part of the response.

Reply:
gotcha. Working on it. Thank you :)

Reply:
Take a look again, please.

Thanks 👍

Reply:
Btw, using `instance.errors` rather than a custom message (You must provide valid fields.), it returns a message like this `<ul class=\"errorlist\"><li>ids<ul class=\"errorlist\"><li>Invalid ID specified.</li></ul></li></ul>`. Let me know which one do we prefer to use?

Reply:
Yup, but we definitely shouldn't return HTML tags there. Maybe some something like `str(instance.errors)`. Also, error dict has methods like `as_data` etc. If we can easily figure out how to get these errors as a message, I would try to do that. Otherwise, we'll merge it as it is, because this is also good enough.

Reply:
cool, checking again. :)